src/main/java/UserManagement.java

```diff
@@ -0,0 +1,112 @@
+import java.util.*;
+
+public class UserManagement {
+    private DataPersistenceService dataPersistenceService = new DataPersistenceService();
+    private List<User> userList = new ArrayList<>();
+    private Map<String, User> userLoginToUserMap = new HashMap<>();
+
+    //TODO: move role logic to a separate file
+    private Map<String, Role> userLoginToRoleMap = new HashMap<>();
```

Comment:
It could be a good idea to use the user id as a key here instead of the user login because an id is something that is immutable (never changes) while login can change (in some cases).

Reply:
Make sense. I'll apply this next time)